internal/pkg/pluginengine/change.go

```diff
@@ -169,3 +182,46 @@ func handleResult(smgr statemanager.Manager, change *Change) error {
 	log.Successf("Plugin %s %s done.", change.Tool.Name, change.ActionName)
 	return nil
 }
+
+// FillInputParams fill inputs from state
+func FillInputParams(smgr statemanager.Manager, options map[string]interface{}) error {
+	//traverse options
+	for key, value := range options {
+		log.Debug("Key: ", key, ", Value: ", value)
+		// judge wheter the value is string
+		if inst, ok := value.(string); ok {
+			// judge wheter the format is ${{}}
+			if strings.HasPrefix(inst, REF_PREFIX) && strings.HasSuffix(inst, REF_SUFFIX) {
+				ref := strings.TrimSpace(strings.TrimSuffix(strings.TrimPrefix(inst, REF_PREFIX), REF_SUFFIX))
+				log.Debug("Ref inputs: ", ref)
+				refParam := strings.Split(ref, ".")
+				if len(refParam) <= 3 {
+					return errors.New("ref input format is not correct: " + ref)
+				}
+
+				state := smgr.GetState(genStateKey(refParam))
+				outputs := state.Resource["outputs"]
+				log.Debug("Ref outputs: ", outputs)
+				if outs, ok := outputs.(map[string]interface{}); ok {
+					log.Debug("Ref outs: ", outs)
+					log.Debug("Ref param: ", refParam[3])
+					options[key] = outs[refParam[3]]
```

Comment:
Magic number. Plus, what if refParam[3] is not in the "outs"?

Reply:
${{ TOOL_NAME.TOOL_KIND.outputs.var}},  when it is splitted to be []string,  element 3 is the var